### DISCUSSION TITLE
sql 審核執行後，狀態一直處在確認中

### DISCUSSION THREAD

**Discussion by dearsiang** [hhyo/Archery#2138] [General]
Hi 

請問sql 審核執行後，狀態一直處在確認中，已經超過24小時，請問有檢查步驟或是解決方法嗎？

這邊有跟SQL提交人確認過，數據庫已根據需求變更完成，但sql 審核執行後，一直處在確認中。
![Uploading Snipaste_2023-04-26_10-01-19.png…]()

**Comment by dearsiang** [comment]
<img width="1150" alt="Snipaste_2023-04-26_10-01-19" src="https://user-images.githubusercontent.com/107904775/234446951-b1a940bd-9c63-4979-b80c-c834d7aed764.png">

**Comment by LeoQuote** [comment]
这个状态代表的意思是工单一直在"执行中" 的状态, 这种情况的话, 一般是工单执行完毕, 但是和 archery 交互失败, 得看这个问题是否经常出现.

如果经常出现的话, 可以看一下服务端的日志, 有没有什么失败的线索, 然后根据情况排查一下.  日志可以通过 docker 或者直接查看文本文件

```
docker logs archery -f --tail=10
logs/archery.log
```

**Reply by surfptcat** [reply]
Hi @LeoQuote 
查看日誌看起來只有連接不到mysql, 是否是因為連不到mysql無法更新執行狀態? 

![Snipaste_2023-04-26_13-35-41](https://user-images.githubusercontent.com/45961171/234479047-f2cbe08f-03f7-4259-a6e2-1e9c165c2f08.jpg)

**Reply by LeoQuote** [reply]
是的没错, 就是没法连接到配置中的 mysql 数据库

**Reply by dearsiang** [reply]
Hi Leo

請問有解決辦法嗎？

**Reply by LeoQuote** [reply]
出现这个错误你得排查一下写的 mysql 地址正确不正确, 如果可以的话, 可以使用外置的 mysql, 不使用 docker-compose 内置的 mysql
这个错误 2003 你也可以自行搜索排查一下, 有没有具体的方案可以用的, 我这边看到要么是鉴权的问题, 要么是连接的主机不符合要求.
